rpc/dcrd.go

```diff
@@ -58,7 +58,8 @@ func (d *DcrdConnect) Close() {
 
 // Client creates a new DcrdRPC client instance. Returns an error if dialing
 // dcrd fails or if dcrd is misconfigured.
-func (d *DcrdConnect) Client(ctx context.Context) (*DcrdRPC, string, error) {
+func (d *DcrdConnect) Client() (*DcrdRPC, string, error) {
+	ctx := context.Background()
```

Comment:
What is the purpose of this context.

If it is to be sure that client can finish what it's doing and cleanly stop in a multi-threaded enviroment, then it needs to take a context and a waitgroup.

If it is to prevent long processes on shutdown from continuing then needs to take a context.

If it is to prevent it from taking to long, it should use `context.WithTimeout`

If it's just to make it work, I think that `context.WithCancel` is recommended, and you do `defer cancel` right after.

In my opinion, the ctx should come from higher up like it was. It can be saved in the struct. All of the rpc calls should take that ctx and further do `context.WithTimeout` with a reasonable timeout, like 5-10 seconds.